### PR TITLE
Expose machine.MarkForRemoval in provisioner API

### PR DESCRIPTION
**Pull request by babbageclunk**
Part of [lp:1585878](https://bugs.launchpad.net/juju-core/+bug/1585878)

Once the machine undertaker is ready the provisioner task will mark machines for removal instead of removing them directly.

From talking to thumper, there's no need to bump the version of the provisioner facade as long as this gets into the beta.

### CODE REVIEW

File: apiserver/provisioner/provisioner_test.go [hunk 1]
```diff
@@ -1235,3 +1235,37 @@ func (s *withoutControllerSuite) TestFindTools(c *gc.C) {
...
+			{Tag: "machine-0-lxd-5"},   // unauthorised
+			{Tag: "application-thing"}, // only machines allowed
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	results := res.Results
+	c.Assert(results, gc.HasLen, 6)
+	c.Check(results[0].Error, gc.IsNil)
+	c.Check(*results[1].Error, gc.Equals,
+		*common.ServerError(errors.NotFoundf("machine 100")))
```
Comment (fwereade) [inline apiserver/provisioner/provisioner_test.go]: Explicitly checking the code (IsCodeNotFound?) is probably best here -- that's what any consumer will actually be responding to. (It's *also* nice to check the message too, but the code is the really important thing.)

**Comment by fwereade** [comment]
It would be really nice if these weren't full-stack tests, but we can't fix everything at once. LGTM with trivially tweaked error-checking.

File: apiserver/provisioner/provisioner_test.go (hunk 1 above)
Comment (dimitern) [inline apiserver/provisioner/provisioner_test.go]: +1

```go
c.Check(results[1].Error, jc.Satisfies, params.IsCodeNotFound)
```

(IIRC)

**Comment by dimitern** [comment]
LGTM as well

File: apiserver/provisioner/provisioner_test.go (hunk 1 above)
Comment (babbageclunk) [inline apiserver/provisioner/provisioner_test.go]: Thanks - added.

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
Ugh, fixed stupid typo. Could've sworn I'd rerun those tests!
$$merge$$

**Pull request merged**